src/main/java/string/StringExercise9.java

```diff
@@ -0,0 +1,28 @@
+package string;
+
+public class StringExercise9 {
+    public String shortcutString(String str) {
+        int i = 0;
+        StringBuilder result = new StringBuilder();
+        while (i < str.length()) {
+            char c = str.charAt(i);
+            int j = i + 1;
+            int count = 1;
+            while (j < str.length()) {
+                char ch = str.charAt(j);
+                if (c == ch)
+                    count++;
+                j++;
+            }
+            if (count > 1) {
+                result.append(c).append(count);
+                i = count + i;
+            }
+            if (count == 1) {
+                result.append(c);
+                i++;
+            }
```

Comment:
Có thể thế này

```suggestion
    result.append(c);
    if(count > 1){
        result.append(count);
     }
     i += count;
```

Với lại, hoàn toàn có thể dùng mỗi biến i không cần biến j ở đây

Reply:
Không dùng biến j vậy vòng lặp while con dùng thế nào anh